internal/pool/pool.go

```diff
@@ -115,15 +115,16 @@ func NewConnPool(opt *Options) *ConnPool {
 }
 
 func (p *ConnPool) checkMinIdleConns() {
-	if p.opt.MinIdleConns == 0 {
+	if p.opt.MinIdleConns == 0 || p.closed() {
```

Comment:
What about moving this from here to `popIdle` where it makes more sense? Something like:

```go
func (p *ConnPool) popIdle() (*Conn, error) {
    if p.closed() {
        return nil, ErrClosed
    }
}
```